pulsar-client-schema/src/test/java/org/apache/pulsar/client/schema/ProtobufSchemaTest.java

```diff
@@ -67,4 +78,33 @@ public void testSchema() {
 
         Assert.assertEquals(schema.toString(), EXPECTED_SCHEMA_JSON);
     }
+
+    @Test
+    public void testGenericOf() {
+        try {
+            ProtobufSchema<org.apache.pulsar.client.schema.proto.Test.TestMessage> protobufSchema
+                    = ProtobufSchema.ofGenericClass(org.apache.pulsar.client.schema.proto.Test.TestMessage.class,
+                    Collections.emptyMap());
+        } catch (Exception e) {
+            Assert.fail();
+        }
+
+        try {
+            ProtobufSchema<org.apache.pulsar.client.schema.proto.Test.TestMessage> protobufSchema
+                    = ProtobufSchema.ofGenericClass(String.class,
+                    Collections.emptyMap());
+            Assert.fail();
```

Comment:
nit: put a meaningful message in fail method 

e.g.
`fail("Should not construct a ProtobufShema over a non-protobuf-generated class")`